web_poet/pages.py

```diff
@@ -63,10 +70,98 @@ def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
 
     async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
+
+        partial_item = await self._partial_item()
+        if partial_item:
+            return partial_item
+
         return await item_from_fields(
             self, item_cls=self.item_cls, skip_nonitem_fields=self._skip_nonitem_fields
         )
 
+    def _get_select_fields(self) -> Optional[SelectFields]:
+        select_fields = getattr(self, "select_fields", None)
```

Comment:
`_get_select_fields()` currently require (and assumes) that the `select_field` instance attribute contains `SelectFields`. We could reflect this in our docs as a requirement.

I think this is the simplest one for now.

---

An alternative would be to support dynamic variable names which could have names other than `select_field`. This means we need to do introspection to find any instances of `SelectFields` while iterating through its instance attributes. This also means that we need to handle edge cases when two or more `SelectFields` are present in a given page. We need a mechanism on how to choose one of them and reflect them in our docs.

This seems to be a more complex and messy approach.

---

Another alternative would be to set `select_fields: Optional[SelectFields]` as a dependency in `ItemPage`. Although I'm not quite sure about this at the moment.

Reply:
Apparently, there's a small problem when adding `ItemPage` as optional:

```python
@attrs.define
class ItemPage(Injectable, Returns[ItemT]):
     select_fields: typing.Optional[SelectFields] = None
     _skip_nonitem_fields: typing.ClassVar[bool]

@attrs.define
class WebPage(ItemPage[ItemT], ResponseShortcutsMixin):
    response: HttpResponse
```

**attrs** raises an exception:

```
ValueError: No mandatory attributes allowed after an attribute with a default value or factory.  Attribute in question: Attribute(name='response', default=NOTHING, validator=None, repr=True, eq=True, eq_key=None, order=True, order_key=None, hash=None, init=True, metadata=mappingproxy({}), type=<class 'web_poet.page_inputs.http.HttpResponse'>, converter=None, kw_only=False, inherited=False, on_setattr=None, alias=None)
```

It also breaks the API for `WebPage(HttpResponse("", b""))` since users would now have to `WebPage(SelectFields(fields={"*": True}), HttpResponse("", b""))`

However, there's a way around it by using `@attrs.define(kw_only=True)`. I think this is much better though since it makes the API more compatible in the future. I'll implement this in the another PR.